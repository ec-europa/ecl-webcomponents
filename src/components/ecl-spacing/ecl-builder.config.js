const path = require('path');
const pkg = require('./package.json');

const nodeModules = path.resolve(__dirname, '../../../node_modules');
const outputFolder = './build';

// SCSS includePaths
const includePaths = [nodeModules];

const banner = `${pkg.name} - ${pkg.version} Built on ${new Date().toISOString()}`;

module.exports = {
  styles: [
    {
      entry: path.resolve(__dirname, 'src/styles/ecl-spacing-ec.scss'),
      dest: path.resolve(outputFolder, 'styles/ecl-spacing-ec.css'),
      options: {
        banner,
        includePaths,
        sourceMap: false,
      },
    },
    {
      entry: path.resolve(__dirname, 'src/styles/ecl-spacing-eu.scss'),
      dest: path.resolve(outputFolder, 'styles/ecl-spacing-eu.css'),
      options: {
        banner,
        includePaths,
        sourceMap: false,
      },
    },
  ],
};
