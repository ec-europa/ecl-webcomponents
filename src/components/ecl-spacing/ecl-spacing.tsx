import { Component, Prop, h, Element } from '@stencil/core';

@Component({
  tag: 'ecl-spacing',
  styleUrls: {
    ec: './build/styles/ecl-spacing-ec.css',
    eu: './build/styles/ecl-spacing-eu.css',
  },
  shadow: false,
  scoped: false,
})

export class EclSpacing {
  @Element() el: HTMLElement;
  @Prop() styleClass: string = '';
  @Prop() theme: string = 'ec';
  @Prop() breakpoint: string;
  @Prop() outer: boolean = true;
  @Prop() inner: boolean = false;
  @Prop() vertical: boolean = false;
  @Prop() horizontal: boolean = false;
  @Prop() direction: string = 'a';
  @Prop() value: string = 'm';

  getClass(): string {
    let baseClass = 'ecl-u-'; 
    let styleClasses = [];

    if (this.outer) {
      if (!this.breakpoint) {
        styleClasses.push(`${baseClass}m${this.direction}-${this.value}`);
      } else {
        styleClasses.push(`${baseClass}m${this.direction}-${this.breakpoint}-${this.value}`);
      }
    } 
    if (this.inner) {
      if (!this.breakpoint) {
        styleClasses.push(`${baseClass}p${this.direction}-${this.value}`);
      } else {
        styleClasses.push(`${baseClass}p${this.direction}-${this.breakpoint}-${this.value}`);
      }
    }
    
    return styleClasses.join(' ');
  }

  render() {
    return (
      <div class={this.getClass()}>
        <slot></slot>
      </div>
    )
  }
}

