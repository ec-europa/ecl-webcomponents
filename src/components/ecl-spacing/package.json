{
  "name": "@ecl/web-components-spacing",
  "author": "European Commission",
  "license": "EUPL-1.2",
  "version": "0.11.0",
  "description": "ECL Spacing",
  "main": "ecl-spacing.ts",
  "private": true,
  "scripts": {
    "build:styles": "cross-env ecl-builder styles",
    "build": "run-s clean build:*",
    "clean": "rimraf build",
    "prepublish": "npm run build"
  },
  "devDependencies": {
    "@ecl/builder": "3.11.0",
    "@ecl/vanilla-utility-spacing": "3.11.0",
    "cross-env": "7.0.3"
  },
  "repository": {
    "type": "git",
    "url": "git+https://github.com/planctus/ecl-webcomponents.git"
  },
  "bugs": {
    "url": "https://github.com/planctus/ecl-webcomponents/issues"
  },
  "homepage": "https://github.com/planctus/ecl-webcomponents",
  "keywords": [
    "ecl",
    "europa-component-library",
    "design-system",
    "web-components"
  ]
}
