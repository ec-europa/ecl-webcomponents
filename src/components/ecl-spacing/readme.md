# ecl-spacing



<!-- Auto Generated Below -->


## Properties

| Property     | Attribute     | Description | Type      | Default     |
| ------------ | ------------- | ----------- | --------- | ----------- |
| `breakpoint` | `breakpoint`  |             | `string`  | `undefined` |
| `direction`  | `direction`   |             | `string`  | `'a'`       |
| `horizontal` | `horizontal`  |             | `boolean` | `false`     |
| `inner`      | `inner`       |             | `boolean` | `false`     |
| `outer`      | `outer`       |             | `boolean` | `true`      |
| `styleClass` | `style-class` |             | `string`  | `''`        |
| `theme`      | `theme`       |             | `string`  | `'ec'`      |
| `value`      | `value`       |             | `string`  | `'m'`       |
| `vertical`   | `vertical`    |             | `boolean` | `false`     |


----------------------------------------------

*Built with [StencilJS](https://stenciljs.com/)*
