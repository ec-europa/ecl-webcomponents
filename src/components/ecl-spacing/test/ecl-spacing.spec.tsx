import { newSpecPage } from '@stencil/core/testing';
import { EclSpacing } from '../ecl-spacing';

describe('ecl-spacing', () => {
  it('margin vertical renders', async () => {
    const page = await newSpecPage({
      components: [EclSpacing],
      html: `<ecl-spacing
        outer
        direction="v"
        value="s"
      ></ecl-spacing>`,
    });

    expect(page.root).toMatchSnapshot();
  });
  it('margin and padding horizontal renders', async () => {
    const page = await newSpecPage({
      components: [EclSpacing],
      html: `<ecl-spacing
        outer
        inner
        direction="h"
        value="l"
      ></ecl-spacing>`,
    });

    expect(page.root).toMatchSnapshot();
  });
  it('margin horizontal auto renders', async () => {
    const page = await newSpecPage({
      components: [EclSpacing],
      html: `<ecl-spacing
        outer
        direction="h"
        value="auto"
      ></ecl-spacing>`,
    });

    expect(page.root).toMatchSnapshot();
  });
  it('margin horizontal none renders', async () => {
    const page = await newSpecPage({
      components: [EclSpacing],
      html: `<ecl-spacing
        outer
        direction="a"
        value="none"
      ></ecl-spacing>`,
    });

    expect(page.root).toMatchSnapshot();
  });
  it('margin horizontal l in xl breakpoint renders', async () => {
    const page = await newSpecPage({
      components: [EclSpacing],
      html: `<ecl-spacing
        outer
        direction="a"
        value="l"
        breakpoint="xl"
      ></ecl-spacing>`,
    });

    expect(page.root).toMatchSnapshot();
  });
});
