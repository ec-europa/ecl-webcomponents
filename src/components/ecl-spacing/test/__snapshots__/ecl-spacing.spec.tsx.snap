// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ecl-spacing margin and padding horizontal renders 1`] = `
<ecl-spacing direction="h" inner="" outer="" value="l">
  <!---->
  <div class="ecl-u-mh-l ecl-u-ph-l"></div>
</ecl-spacing>
`;

exports[`ecl-spacing margin horizontal auto renders 1`] = `
<ecl-spacing direction="h" outer="" value="auto">
  <!---->
  <div class="ecl-u-mh-auto"></div>
</ecl-spacing>
`;

exports[`ecl-spacing margin horizontal l in xl breakpoint renders 1`] = `
<ecl-spacing breakpoint="xl" direction="a" outer="" value="l">
  <!---->
  <div class="ecl-u-ma-xl-l"></div>
</ecl-spacing>
`;

exports[`ecl-spacing margin horizontal none renders 1`] = `
<ecl-spacing direction="a" outer="" value="none">
  <!---->
  <div class="ecl-u-ma-none"></div>
</ecl-spacing>
`;

exports[`ecl-spacing margin vertical renders 1`] = `
<ecl-spacing direction="v" outer="" value="s">
  <!---->
  <div class="ecl-u-mv-s"></div>
</ecl-spacing>
`;
